packages/runtime/container-runtime/src/opLifecycle/opGroupingManager.ts

```diff
@@ -49,7 +49,14 @@ export class OpGroupingManager {
 		this.logger = createChildLogger({ logger, namespace: "OpGroupingManager" });
 	}
 
-	public groupBatch(batch: IBatch): IBatch {
+	/**
+	 * Converts the given batch into a "grouped batch" - a batch with a single message of type "groupedBatch",
+	 * with contents being an array of the original batch's messages.
+	 *
+	 * @remarks - Remember that a BatchMessage has its content JSON serialized, so the incoming batch message contents
+	 * must be parsed first, and then the type and contents mentioned above are hidden in that JSON serialization.
+	 */
+	public groupBatch(batch: IBatch): IBatch<[BatchMessage]> {
```

Comment:
I think you should get a compiler error that you should drop default value of templated argument.
Even if you do not, it's weird that on the same like, you do use default value and explicitly name the type (i.e. IBatch vs. ```IBatch<[BatchMessage]>```

Reply:
Maybe it's too subtle, but this is saying it's an `IBatch` of a singleton array.  I actually did have the default value in the other `IBatch` instance for contrast but had to remove it due to the compiler error :(